enterprise/api/v1/views.py

```diff
@@ -835,3 +835,23 @@ def post(self, request):
                 {'error': str('Request codes email could not be sent')},
                 status=HTTP_500_INTERNAL_SERVER_ERROR
             )
+
+
+class TableauAuthViewSet(generics.GenericAPIView):
+    """
+    API to authenticate user with Tableau.
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        """
+        Get the auth token against logged in user from tableau
+        """
+        url = settings.TABLEAU_URL + '/trusted'
+        payload = {'username': request.user.username}
```

Comment:
should this user be a service user instead of request user? would there be a separate token for each user?

Reply:
Yes, there will be a separate token for each user.